causync.py

```diff
@@ -22,51 +22,77 @@ class CauSync(object):
             task (str): contains the task to execute
             no_incremental (bool): if True, CauSync doesn't do incremental backups
             quiet (bool): if set to True, CauSync doesn't print anything to console
-            selfname (string): the program's name, should be copied from sys.argv[0], used by is_running()
+            dry_run (bool): same as rsync's -n argument, does a dry run
+            selfname (str): the program's name, should be copied from sys.argv[0], used by is_running()
+            excludes(str|list): string or list of files to exclude
+            exclude_from(str): exclude file
+            loglevel(int): logging level (see config or help(logging)
 
         Attributes:
             pid (int): PID of the current process
-            src_abs (string): absolute path of source directory
-            dst_abs (string): absolute path of destination directory, joined with date_ival
+            src_abs (list): list containing source directory absolute paths
+            dst_abs (str): absolute path of destination directory, joined with date_ival
+            curdate (datetime): datetime object containing the current date
             logger (object): the logger object used for logging to file/console
     """
 
     curdate = None
+    multiple_sources = False
 
-    def __init__(self, config, src, dst, task, no_incremental=False, quiet=False, dry_run=False, selfname="causync.py"):
+    def __init__(self, config, src, dst, task, no_incremental=False, quiet=False,
+                 dry_run=False, selfname="causync.py", excludes=None, exclude_from=False,
+                 loglevel=None):
         self.config = config
         self.name = selfname
         self.pid = os.getpid()
         self.no_incremental = no_incremental
+        self.excludes = excludes if excludes else []
+        if exclude_from:
+            self.excludes = self.excludes + self.parse_exclude_file(exclude_from)
 
         self.src = src
-        self.src_abs = os.path.abspath(self.src)
+        self.src_abs = self.parse_src(self.src)
+        if not self.src_abs:
+            exit(-1)
+
         self.dst = dst
         self.dst_abs = os.path.abspath(dst)
 
         self.task = task
         self.quiet = quiet
         self.dry_run = dry_run
         self.curdate = datetime.now()
-        self.logger = self.get_logger()
+        self.logger = self.get_logger(loglevel)
+
+    def parse_src(self, src):
+        if type(src) == type(list()):
+            self.multiple_sources = True if len(src) > 1 else False
+            return [os.path.abspath(i) for i in src]
+        elif type(self.src) == str:
+            return [os.path.abspath(self.src)]
+        else:
+            self.logger.error("source directory type error")
+            return False
 
     def run(self):
         """ Main run function. """
 
         is_running = self.is_running()
-        lockfile_exists = self.lockfile_exists()
+        lockfile_exists = self.check_lockfiles()
```

Comment:
As I see, you can have multiple lock files if multiple source directories are given. Please describe in the README how this works in this case.